### PR TITLE
Suryansh/frontendtestcases

### PR DESCRIPTION
**Pull request by suryansh-max**
1. added test properly
2. changed bordgui to handle clicks properly

### CODE REVIEW

File: frontend/BoardGUI.py [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import pygame as pg
-# from backend.Board import Board
+from backend.Board import Board
```
Comment (raspberrymilkyway) [inline frontend/BoardGUI.py:2]: I think it'd be better to keep the Board class out of this and just use the GameManager class to call all backend classes and functions.

If you're looking to use the board in the tests, we can make up a function to print board or check its status or something. I think board_gui.backend_game.board is also equivalent.

As it is, the import in line 2 is only being used to create a board in line 15, which never gets used.

**Comment by EXC3ll3NTrhyTHM** [comment]
You need to pull the latest changes from main and rerun your tests. The BoardGUI has changes in it that are outside of the scope of creating the frontend tests.

File: frontend/BoardGUI.py (hunk 1 above)
Comment (suryansh-max) [reply frontend/BoardGUI.py:2]: may be i can remove that particular test which uses that, and that can be refactored once we are done with game logic

**Comment by suryansh-max** [comment]
created another pull request with latest main

**Pull request closed without merging**